### PR TITLE
RDM-7165: DynamicRadioList and DynamicMultiSelectList

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-7165

### Change description ###
DynamicRadioList and DynamicMultiSelectList

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/aat/resources/features/F-121/S-121.6.td.json [hunk 1]
```diff
@@ -772,14 +774,15 @@
...
-								},
+									"retain_hidden_value": null,
+								"retain_hidden_value": null},
 								{
 									"complex_field_element_id": "RespondentDetails.RespAddress3.County",
 									"display_context": "HIDDEN",
 									"label": null,
 									"hint_text": null,
 									"show_condition": null,
+                  "retain_hidden_value": null,
```
Comment (mario-paniccia) [inline src/aat/resources/features/F-121/S-121.6.td.json:785]: how come there's 2 `retain_hidden_value`?

File: src/aat/resources/features/F-121/S-121.2.td.json [hunk 2]
```diff
@@ -647,49 +647,56 @@
                   "display_context": "MANDATORY",
                   "label": null,
                   "hint_text": null,
-                  "show_condition": null
+                  "show_condition": null,
```
Comment (mario-paniccia) [inline src/aat/resources/features/F-121/S-121.2.td.json:650]: plenty of duplication here. Can you use Abi's approach to remove duplication in test data please?

File: src/aat/resources/features/F-121/S-121.6.td.json (hunk 1 above)
Comment (Olawale) [reply src/aat/resources/features/F-121/S-121.6.td.json]: Looks like this happened during a merge. Resolved.

File: src/aat/resources/features/F-121/S-121.2.td.json (hunk 2 above)
Comment (Olawale) [reply src/aat/resources/features/F-121/S-121.2.td.json:650]: Will raise a PR off develop branch to fix this.

**Review by mario-paniccia — APPROVED** [review]
👍

**Pull request merged**